packages/analytics-js-integrations/src/integrations/MoEngage/browser.js

```diff
@@ -1,128 +1,180 @@
+    // Check if the user id is same as previous session if not a new session will start
+    if (!rudderElement.message) {
+      logger.error('Payload not correct');
+      return;
+    }
+    const { event, properties, userId } = rudderElement.message;
+    if (userId && this.initialUserId !== userId) {
+      this.reset();
+    }
+    // track event : https://docs.moengage.com/docs/tracking-events
+    if (!event) {
+      logger.error('Event name is not present');
+      return;
+    }
+    if (properties) {
+      window.Moengage.track_event(event, properties);
+    } else {
+      window.Moengage.track_event(event);
+    }
+  }
+
+  reset() {
+    console.log(window.moe, window.moeBannerText, window.Moengage);
+    this.initialUserId = this.analytics.getUserId();
+    window.Moengage.destroy_session()
+  }
+
+  identify(rudderElement) {
+    const { userId, context } = rudderElement.message;
+    let traits = null;
+    if (context) {
+      traits = context.traits;
+    }
+
+    // We are destroying the session if userId is changed or initialUserId is not empty and userId is empty
+    // This is to ensure that if the user logs out and logs in again, we reset it
+    if (
+      (this.initialUserId !== '' && this.initialUserId !== userId) ||
+      (this.initialUserId !== '' && userId === '')
+    ) {
+      this.reset();
+    }
+
+    // If initialUserId is empty, set it to the current userId this happens when an anonymous user loggedIn for the first time
+    if (this.initialUserId === '' && userId) {
+      this.initialUserId = userId;
+    }
+
+    const userAttributes = {};
+    each((value, key) => {
+      if (Object.prototype.hasOwnProperty.call(identifyUserPropertiesMap, key)) {
+        const method = identifyUserPropertiesMap[key];
+        userAttributes[method] = value;
+      }
+    }, traits);
+
+    let idPayload = {};
+    if (userId !== '') {
+      idPayload = { uid: userId, anonymousId: this.analytics.getAnonymousId() };
+    } else {
+      idPayload = { anonymousId: this.analytics.getAnonymousId() };
+    }
+
+    // It may happen that userId is not present so we will use anonymousId as uid
+    // https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/supported-api/#set-a-blank-user-id
+    const payload = {
+      // uid: userId ? userId : this.analytics.getAnonymousId(),
+      ...idPayload,
+      ...userAttributes,
+    };
+
+    window.Moengage.identifyUser(payload);
+
+
+    // track user attributes : https://docs.moengage.com/docs/tracking-web-user-attributes
+
+    if (traits) {
+      each((value, key) => {
+        // check if name is present
+        if (key === 'name') {
+          window.Moengage.add_user_name(value);
+        }
+        if (Object.prototype.hasOwnProperty.call(traitsMap, key)) {
+          const method = `add_${traitsMap[key]}`;
+          window.Moengage[method](value);
+        } else {
+          window.Moengage.add_user_attribute(key, value);

```

Comment:
Is the old implementation still required ? what attributes are getting added here apart from [userAttributes](https://github.com/rudderlabs/rudder-sdk-js/pull/2367/files?w=1#diff-50d29403c6a7d3595095d7efce8556d6f7c10f9e5ac00f1556c718a0b093ee42R155) map